use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,153 @@
+virtio queue sharing with a user space process on the same host. So SPDK can expose
+vhost device via vhost-user protocol.
+
+SPDK vhost is a vhost-user slave server. It exposes Unix domain sockets and
+allows external applications to connect. It is capable of exposing virtualized
+storage devices to QEMU instances or other arbitrary processes.
+
+Currently, SPDK vhost-user target can exposes these types of virtualized devices:
+
+- `vhost-user-blk`
+- `vhost-user-scsi`
+- `vhost-user-nvme`
+
+For more information, visit [SPDK](https://spdk.io) and [SPDK vhost-user target](https://spdk.io/doc/vhost.html).
+
+## Install and setup SPDK vhost-user target
+
+### Get source code and build SPDK
+
+Following SPDK [getting started](https://spdk.io/doc/getting_started.html)
+
+### Run SPDK vhost-user target
+
+First, run the SPDK setup.sh script to setup some hugepages for the SPDK vhost
+target application. This will allocate 4096MiB (4GiB) of hugepages, enough for
+the SPDK vhost target and the virtual machine.
+
+```bash
+$ sudo HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory `/var/run/kata-containers/vhost-user` as Kata's vhost-user
+device directory. Make subdirectories for vhost-user sockets and device nodes.
+
+For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+```bash
+mkdir -p /var/run/kata-containers/vhost-user/
+mkdir -p /var/run/kata-containers/vhost-user/block/
+mkdir -p /var/run/kata-containers/vhost-user/block/sockets/
+mkdir -p /var/run/kata-containers/vhost-user/block/devices/
+```
+
+Next, start the SPDK vhost target application.  The following command will start
+vhost on CPU cores 0 with all future socket files placed in `/var/run/kata-containers/vhost-user/block/sockets/`.
+
+```bash
+$ sudo app/spdk_tgt/spdk_tgt -S /var/run/kata-containers/vhost-user/block/sockets/ &
+```
+
+To list all available vhost options use the following command.
+
+```bash
+$ app/spdk_tgt/spdk_tgt -h
+```
+
+Create a experimental `vhost-user-blk` device based on memory directly.
+
+- The following RPC will create a 64MB memory block device with 4096-byte block
+size.
+
+```bash
+$ sudo scripts/rpc.py bdev_malloc_create 64 4096 -b Malloc0
+```
+
+- The following RPC will create a `vhost-user-blk` device exposing Malloc0 block
+device. The device will be accessible via
+`/var/run/kata-containers/vhost-user/block/sockets/vhostblk0`.
+
+```bash
+$ sudo scripts/rpc.py vhost_create_blk_controller vhostblk0 Malloc0
+```
+
+## Host setup for vhost-user devices
+
+Considering OCI spec and characteristic of vhost-user device, Linux reserved
+block major range (240-254) to map each vhost-user block type to a major。
+Also a specific directory is used for vhost-user devices. The base directory is
+a configurable value, with the default being `/var/run/kata-containers/vhost-user`.
+It can be configured by parameter `vhost_user_store_path` in Kata TOML
+configuration file.
```

Comment:
it would be nice to point to https://github.com/kata-containers/runtime/blob/master/README.md#configuration, what do you think ?